sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosPagedIterableTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ */
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.InternalObjectNode;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.rx.TestSuiteBase;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CosmosPagedIterableTest extends TestSuiteBase {
+
+    private static final int NUM_OF_ITEMS = 10;
+
+    private CosmosClient cosmosClient;
+    private CosmosContainer cosmosContainer;
+
+    @Factory(dataProvider = "clientBuilders")
```

Comment:
ditto we should use client builder with at least session, otherwise the test will be flaky.

perhaps `clientBuildersWithDirectSession`